src/main/java/uk/gov/companieshouse/api/accounts/controller/DebtorsController.java

```diff
@@ -72,15 +73,26 @@ public ResponseEntity create(@Valid @RequestBody Debtors debtors,
 
         } catch (DataException ex) {
 
-            final Map<String, Object> debugMap = new HashMap<>();
-            debugMap.put(TRANSACTION_ID, transaction.getId());
-            debugMap.put(COMPANY_ACCOUNT_ID, companyAccountId);
+            final Map<String, Object> debugMap = createDebugMap(companyAccountId, transaction);
             debugMap.put(MESSAGE, "Failed to create debtors resource");
             LOGGER.errorRequest(request, ex, debugMap);
             responseEntity = apiResponseMapper.map(ex);
+        } catch (RestException re) {
+
+            final Map<String, Object> debugMap = createDebugMap(companyAccountId, transaction);
+            debugMap.put(MESSAGE, "Failed to get company profile in validation for company" + transaction.getCompanyNumber());
```

Comment:
This `debugMap.put(MESSAGE, "Failed to get company profile in validation for company" + transaction.getCompanyNumber());` could be moved into your createDebugMap method, and then pass the message in as an arguement. This will mean all the puts can be located together.